lib/msal-browser/test/app/PublicClientApplication.spec.ts

```diff
@@ -1789,4 +1791,80 @@ describe("PublicClientApplication.ts Class Unit Tests", () => {
             expect(account).to.be.null;
         });
     });
+
+    describe("broadcastEvent and addEventCallback tests", () => {
+        it("can add an event callback and broadcast to it", () => {
+            let receivedMessage: BroadcastMessage;
+            const subscriber = (message) => {receivedMessage = message};
+            pca.addEventCallback(subscriber);
+            pca.broadcastEvent(BroadcastEvent.LOGIN_START, InteractionType.POPUP);
+
+            expect(receivedMessage.type).to.deep.eq(BroadcastEvent.LOGIN_START);
+            expect(receivedMessage.interactionType).to.deep.eq(InteractionType.POPUP);
```

Comment:
For tests with callbacks like this, it is often better to use the `done` callback that is passed into the test, and invoke that in the callback when the message is received. This is because with tests that rely on callbacks or async behavior, that the test can complete before the `expect` statements are running, potentially resulting in a false positive.

```js
it("can add an event callback and broadcast to it", (done) => {
  const subscriber = (message) => {
    expect(message.type).to.deep.eq(BroadcastEvent.LOGIN_START);
    expect(message.interactionType).to.deep.eq(InteractionType.POPUP);
    done();
  }
  
  pca.addEventCallback(subscriber);
  pca.broadcastEvent(BroadcastEvent.LOGIN_START, InteractionType.POPUP);
});
```

Reply:
Makes sense. Modified.